src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -332,41 +271,21 @@ default ClassDesc componentType() {
      * default package, or this {@linkplain ClassDesc} does not describe a class or interface type
      */
     default String packageName() {
-        if (!isClassOrInterface())
-            return "";
-        String desc = descriptorString();
-        int index = desc.lastIndexOf('/');
-        return (index == -1) ? "" : internalToBinary(desc.substring(1, index));
+        return "";
     }
 
     /**
      * Returns a human-readable name for the type described by this descriptor.
      *
      * @implSpec
-     * <p>The default implementation returns the simple name
+     * <p>The implementations return the simple name
```

Comment:
```suggestion
     * <p>The implementation returns the simple name
```

I think this can be singular as it's not required to have more than 1 implementation class.

Reply:
Hmm, in fact, should we still have this `@implSpec` if we make this default?

Reply:
I think this can be dropped as `ClassDesc` as well.   It can say "This method returns....".   See what CSR review would say.